lms/djangoapps/experiments/stable_bucketing.py

```diff
@@ -22,11 +22,14 @@ def stable_bucketing_hash_group(group_name, group_count, username):
     Arguments:
         group_name: The name of the grouping/experiment.
         group_count: How many groups to bucket users into.
-        username: The username of the user being bucketed.
+        user: The user being bucketed.
     """
+    # We need username for stable bucketing and id for tracking, so just skip anonymous (not-logged-in) users
+    if not user or not user.id:
+        return 0
```

Comment:
This means we can never bucket anonymous users, presumably? i.e., ExperimentWaffleFlag will only do anything with authenticated users?

Reply:
Yeah ExperimentWaffleFlag currently comes back as disabled if the user is anonymous, so we're never bucketing anonymous users. This change is happening because we still want anonymous users to have access to the MFE, but there's still no need for them to be bucketed into a control group. We're just moving the check around to allow force-bucket flags that are on for everyone to work as expected (to actually be on for _everyone_ and not just authenticated users)